openstack-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1009 @@
+      + table.panelOptions.withDescription('Detailed view of the routers managed by Neutron.')
+      + table.fieldConfig.defaults.custom.withAlign('center')
+      + table.standardOptions.withOverridesMixin([
+        table.fieldOverride.byRegexp.new('/Admin up|Status/')
+        + table.fieldOverride.byRegexp.withProperty('custom.displayMode', 'color-text')
+        + table.fieldOverride.byRegexp.withPropertiesFromOptions(
+          table.standardOptions.withMappings(
+            {
+              type: 'value',
+              options: {
+                ACTIVE: {
+                  color: 'green',
+                  index: 2,
+                  text: 'Active',
+                },
+                DOWN: {
+                  color: 'red',
+                  index: 3,
+                  text: 'Down',
+                },
+                'false': {
+                  color: 'red',
+                  index: 1,
+                  text: 'False',
+                },
+                'true': {
+                  color: 'green',
+                  index: 0,
+                  text: 'True',
+                },
+              },
+            }
+          ),
+        ),
+      ])
+      + table.queryOptions.withTransformationsMixin([
+        {
+          id: 'joinByField',
+          options: {
+            byField: 'Time',
+            mode: 'outer',
+          },
+        },
+        {
+          id: 'organize',
+          options: {
+            excludeByName: {
+              Time: true,
+              Value: true,
+              __name__: true,
+              hostname: true,
+              job: true,
+              project_id: true,
+            },
+            indexByName: {
+              name: 0,
+              instance: 1,
+              status: 2,
+              admin_state_up: 3,
+              id: 4,
+              external_network_id: 5,
+              project_id: 6,
+              Time: 7,
+              __name__: 8,
+              job: 9,
+              Value: 10,
+            },
+            renameByName: {
+              Time: '',
+              adminState: 'Admin state',
+              admin_state_up: 'Admin up',
+              external_network_id: 'External network ID',
+              id: 'ID',
+              instance: 'Instance',
+              name: 'Name',
+              project_id: '',
+              service: 'Service',
+              status: 'Status',
+              zone: 'Zone',
+            },
+            includeByName: {},
+          },
+        },
+      ]),
+
+    ports:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Ports',
+        targets=[t.ports, t.portsLBNotActive, t.portsNoIPs],
+        description='The number of routers managed by Neutron.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    portDetails:
+      table.new('Port details')
+      + table.queryOptions.withTargets([t.portDetails])
+      + table.queryOptions.withDatasource('prometheus', '${prometheus_datasource}')
```

Comment:
Nit, but can you please ensure this panel matches the width of the others? Or make it take up 2/3rds instead of just over half (e.g. match it to the VMs panel in size if you can't fit it on half the screen width)
![image](https://github.com/grafana/jsonnet-libs/assets/1282515/48290d49-b9f2-4054-96aa-668ef743c767)